src/libs/ajax/workflows-app/Cbas.test.js

```diff
@@ -0,0 +1,337 @@
+      ],
+      uponReceiving: 'post a run set with a "none" source',
+      withRequest: { path: '/api/batch/v1/run_sets', method: 'POST', ...jsonBody(payload) },
+      willRespondWith: { status: 200, body: expectedResponse },
+    });
+
+    await cbasPact.executeTest(async (mockService) => {
+      // ARRANGE
+      const signal = 'fakeSignal';
+
+      fetchOk.mockImplementation(async (path) => await fetch(`${mockService.url}/${path}`, { method: 'POST', ...jsonBody(payload) }));
+      fetchFromProxy.mockImplementation(() => fetchOk);
+
+      // ACT
+      const response = await Cbas(signal).runSets.post(mockService.url, payload);
+
+      // ASSERT
+      expect(response).toBeDefined();
+      expect(fetchOk).toBeCalledTimes(1);
+      expect(fetchFromProxy).toBeCalledTimes(1);
+      expect(fetchOk).toBeCalledWith('api/batch/v1/run_sets', { method: 'POST', ...jsonBody(payload), signal });
+      expect(response).toHaveProperty('run_set_id');
+      expect(response.runs.length).toEqual(1);
+    });
+  });
+
+  it('should successfully POST a run_set containing a struct input', async () => {
+    const expectedResponse = {
+      run_set_id: fromProviderState('${run_set_id}', '00000000-0000-0000-0000-000000000000'), // eslint-disable-line no-template-curly-in-string
+      runs: [
+        {
+          run_id: fromProviderState('${run_id}', '00000000-0000-0000-0000-000000000000'), // eslint-disable-line no-template-curly-in-string
+          state: regex(RUN_STATE_REGEX, 'RUNNING'),
+          errors: regex('.*', 'some arbitrary string'),
+        },
+      ],
+      state: regex(RUNSET_STATE_REGEX, 'RUNNING'),
+    };
+
+    const payload = {
+      run_set_name: 'myRunSet',
+      run_set_description: 'myRunSet description',
+      method_version_id: '90000000-0000-0000-0000-000000000009',
+      wds_records: { record_type: 'FOO', record_ids: ['FOO1'] },
+      workflow_input_definitions: runSetInputDefWithStruct,
+      workflow_output_definitions: runSetOutputDef,
+    };
+
+    await cbasPact.addInteraction({
+      states: [
+        { description: 'ready to fetch recordId FOO1 from recordType FOO from wdsService' },
+        { description: 'ready to fetch myMethodVersion with UUID 90000000-0000-0000-0000-000000000009' },
+        { description: 'ready to receive exactly 1 call to POST run_sets' },
+      ],
+      uponReceiving: 'post a run set with a struct source',
+      withRequest: { path: '/api/batch/v1/run_sets', method: 'POST', ...jsonBody(payload) },
+      willRespondWith: { status: 200, body: expectedResponse },
+    });
+
+    await cbasPact.executeTest(async (mockService) => {
+      // ARRANGE
+      const signal = 'fakeSignal';
+
+      fetchOk.mockImplementation(async (path) => await fetch(`${mockService.url}/${path}`, { method: 'POST', ...jsonBody(payload) }));
+      fetchFromProxy.mockImplementation(() => fetchOk);
+
+      // ACT
+      const response = await Cbas(signal).runSets.post(payload);
```

Comment:
Post body expects proxy url as first argument
```suggestion
      const response = await Cbas(signal).runSets.post(mockService.url, payload);
```